inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,220 @@
+                arxiv_categories.append(arxiv_id.split('/')[0])
+
+        builder.add_arxiv_eprint(
+            arxiv_id=form_fields.get('arxiv_id'),
+            arxiv_categories=arxiv_categories
+        )
+
+    if form_fields.get('doi'):
+        builder.add_doi(doi=form_fields.get('doi'))
+
+    if form_fields.get('subject_term'):
+        builder.add_inspire_categories(
+            subject_terms=form_fields.get('subject_term'),
+            source='user'
         )
-        if title == title_crossref:
-            data['titles'][0]["source"] = "CrossRef"
-        else:
-            data['titles'].append({
-                'title': title_crossref,
-                'source': "CrossRef"
-            })
+
+    if form_fields.get('extra_comments'):
+        builder.add_private_notes(form_fields.get('extra_comments'))
+
+    if form_fields.get('nonpublic_note'):
+        builder.add_private_notes(form_fields.get('nonpublic_note'))
+
+    if form_fields.get('hidden_notes'):
+        builder.add_private_notes(form_fields.get('hidden_notes'))
+
+    if form_fields.get('conf_name'):
+        builder.add_private_notes(form_fields.get('conf_name'))
+
+    # journal year
+    year = form_fields.get('year')
     try:
-        data['titles'][0]['source']
-    except KeyError:
-        # Title has no source, so should be the submitter
-        data['titles'][0]['source'] = "submitter"
-
-    # ============================
-    # Conference name
-    # ============================
-    if 'conf_name' in form_fields:
-        if 'nonpublic_note' in form_fields:
-            data.setdefault("hidden_notes", []).append({
-                "value": form_fields['conf_name']
-            })
-            data['hidden_notes'].append({
-                'value': form_fields['nonpublic_note']
-            })
-        else:
-            data.setdefault("hidden_notes", []).append({
-                "value": form_fields['conf_name']
-            })
-        data['collections'].extend([{'primary': "ConferencePaper"}])
-
-    # ============================
-    # Page number
-    # ============================
-    if 'page_nr' not in data:
-        first_publication_info = data.get('publication_info', [{}])[0]
-
-        page_start = first_publication_info.get('page_start')
-        page_end = first_publication_info.get('page_end')
-
-        if page_start and page_end:
+        year = int(year)
+    except (TypeError, ValueError):
+        year = None
+
+    # journal title
+    if form_fields.get('journal_title'):
+        hits = JournalsSearch().query(
+            'match', title_variants__title__lowercased=form_fields['journal_title']).execute()
+
+        if hits:
             try:
-                data['page_nr'] = int(page_end) - int(page_start) + 1
-            except (TypeError, ValueError):
+                form_fields['journal_title'] = hits[0].short_titles[0].title
+            except (AttributeError, IndexError):
```

Comment:
This should be factored out and put in some utilities. I.e. something like: `normalize_journal_title()`. @jacquerie and @jmartinm were musing some days ago about a reorganization of such utilities.